### PR TITLE
contrib/net/http: Allow passthrough of http trace span options

### LINKED ISSUE #428: contrib/net/http: Allow passthrough of http trace span options
I would like to add additional tags to the traced http span.

Currently I've written a piece of middleware to leverage the `http.WrapHandler` func that is provided.  However, it does not accept a `spanopts` param which the underlying `httputil.TraceAndServce` function takes.  This function is under `/internal` so I cannot use it directly.

I was hoping we could update the signature of `WrapHandler` to include an additional `spanopts ...ddtrace.StartSpanOption` that can be passed through.

Since this is a variadic argument it should be backwards compatible.

Thoughts?  I have a pull request ready...  :)

### PR DESCRIPTION
**Pull request by cwaeland**
* Adds a new `spanopts ...ddtrace.StartSpanOption` parameter to the `WrapHandler` func.

Fixes #428

### CODE REVIEW

File: contrib/net/http/option.go [hunk 1]
```diff
@@ -16,6 +16,9 @@ type config struct {
 // MuxOption represents an option that can be passed to NewServeMux.
```
Comment (gbbr) [inline contrib/net/http/option.go]: Can we add a comment here saying that this is now deprecated in favor of `Option`?

File: contrib/net/http/option.go [hunk 2]
```diff
@@ -44,6 +47,7 @@ func WithAnalyticsRate(rate float64) MuxOption {
 	}
 }
 
+// WithSpanOptions sets additional options for the traced Span
```
Comment (gbbr) [inline contrib/net/http/option.go]: Would you mind making this a bit clearer? e.g.
```go
// WithSpanOptions defines a set of additional ddtrace.StartSpanOption to be added
// to spans started by the integration.
```

File: contrib/net/http/http.go [hunk 3]
```diff
@@ -13,13 +13,13 @@ import (
...
 	*http.ServeMux
-	cfg *muxConfig
+	cfg *config
 }
 
 // NewServeMux allocates and returns an http.ServeMux augmented with the
 // global tracer.
-func NewServeMux(opts ...MuxOption) *ServeMux {
-	cfg := new(muxConfig)
+func NewServeMux(opts ...Option) *ServeMux {
```
Comment (gbbr) [inline contrib/net/http/http.go]: `ServeMux` should also benefit from the new option, let's just change [line 41](https://github.com/DataDog/dd-trace-go/blob/7515d751c02467b60c9044140db3b7cace04c44f/contrib/net/http/http.go#L41) to be:
```go
opts := mux.cfg.spanOpts
```

**Comment by cwaeland** [comment]
@gbbr Alright, updated to address your feedback.

I'm not sure why CircleCI is failing though.  Seems to be failing during the fetch dependency step.

**Comment by gbbr** [comment]
I restarted CI. Let's see. Maybe it was just a flake.

File: contrib/net/http/option.go [hunk 4]
```diff
@@ -7,22 +7,26 @@ import (
...
-type muxConfig struct {
+type config struct {
 	serviceName   string
 	analyticsRate float64
+	spanOpts      []ddtrace.StartSpanOption
 }
 
-// MuxOption represents an option that can be passed to NewServeMux.
-type MuxOption func(*muxConfig)
+// DEPRECATED: Use Option in favor of MuxOption.
```
[the remark this answers is not in the document]
Comment (cwaeland) [reply contrib/net/http/option.go]: I have no idea! 😅

**Comment by cwaeland** [comment]
Hmm, no luck :(

**Comment by cwaeland** [comment]
Any luck?  Was hoping to get it merged today since we're blocked on our side.

Doesn't seem like it should be related to my change.

**Comment by gbbr** [comment]
I will get this in and merged tomorrow. If there are any further changes needed, I will do them myself in this PR. You can feel free to use this branch if you're eager. This will be the final API, there will be no changes in that respect.

**Comment by gbbr** [comment]
Also, not sure if this helps, but you can also simply get the span in your handler and make any changes you want to it automatically:

```go
// r *http.Request
if span, ok := tracer.SpanFromContext(r.Context()); ok {
    // use span however you wish
    // ...
}
```

**Comment by cwaeland** [comment]
Awesome thank you!

**Comment by gbbr** [comment]
You have an error in your code:
```
# gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http
contrib/net/http/http.go:8:2: imported and not used: "gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
```
It's failing CI.

**Comment by gbbr** [comment]
> Awesome thank you!

You're welcome. You can rest assured this will be in by the time you come back to work tomorrow. The API is final.

**Comment by cwaeland** [comment]
Not sure how I missed the import.  My IDE usually catches it.

**Comment by gbbr** [comment]
@cwaeland this has been released now as part of [1.13.0](https://github.com/DataDog/dd-trace-go/releases/tag/v1.13.0)

**Comment by cwaeland** [comment]
Awesome, thanks for getting back to me!

**Pull request merged**